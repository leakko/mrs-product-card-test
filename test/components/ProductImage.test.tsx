import React from "react";
import renderer from "react-test-renderer";
import { ProductCard } from "../../src";
import { product2 } from "../data/products";

describe('ProductImage', () => {
  test('should show component with customized image', () => {
    const wrapper = renderer.create(
        <ProductCard.Image img="./custom-imgage.png"  className="custom-class" />
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show component with product name', () => {
    const wrapper = renderer.create(
        <ProductCard product={product2}>
            {
                () => (
                    <ProductCard.Image />
                )
            }
        </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
})

