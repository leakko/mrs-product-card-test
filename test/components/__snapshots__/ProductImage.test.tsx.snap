// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should show component with customized image 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="./custom-imgage.png"
/>
`;

exports[`ProductImage should show component with product name 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
